src/common/config_validator.cpp

```diff
@@ -0,0 +1,142 @@
+#include "common/config_validator.h"
+#include "3rd_party/exception.h"
+#include "common/logging.h"
+#include "common/regex.h"
+#include "common/utils.h"
+
+#include <boost/filesystem.hpp>
+
+namespace marian {
+
+bool ConfigValidator::has(const std::string& key) const {
```

Comment:
Why would a validator be asked whether it has a key? Please add a comment. If this is a priviate method, it may clarify to add a comment /*private*/

Reply:
The class was just extracted from the old ConfigParser to make removing boost::program_options easier.